airflow/api/common/experimental/mark_tasks.py

```diff
@@ -191,34 +192,121 @@ def set_state(task, execution_date, upstream=False, downstream=False,
     return tis_altered
 
 
-def set_dag_run_state(dag, execution_date, state=State.SUCCESS, commit=False):
+@provide_session
+def set_dag_run_state_to_success(dag, execution_date, commit=False,
+                                 session=None):
     """
-    Set the state of a dag run and all task instances associated with the dag
-    run for a specific execution date.
+    Set the dag run for a specific execution date and its task instances
```

Comment:
I'm a bit late to the party. 

Nit-pick:  Maybe we can factor out the common code get rid of the dupe code. Something like:

```
def set_dag_run_state(dag, execution_date, to_state, from_states=[], commit=False, session=None):
    ...

def set_dag_run_state_to_success(dag, execution_date, commit=False, session=None):
    set_dag_run_state(dag, execution_date, to_state=State.SUCCESS, commit=commit, session=session)

def set_dag_run_state_to_failed(dag, execution_date, commit=False, session=None):
    set_dag_run_state(dag, execution_date, to_state=State.FAILED, from_states=[State.RUNNING], commit=commit, session=session)
````

Reply:
The code was originally like that and I updated to the current form per request from @aoen and @saguziel because "Using if conditions based on param types is kind of an antipattern". I agree with their point since the diff in the logic is big enough.

Reply:
I have a different opinion, but not strong enough to insist on it. Majority rules so lgtm :P

Reply:
@yrqls21, I kind of agree with @jgao54 in this case. Looking at the logic of the 3 functions (`to_success`,`to_failed`,`to_running`), the amount of duplicated code is significant. They all have the same signature, and could be refactored into one as suggested by Joy. See [here](https://refactoring.com/catalog/parameterizeMethod.html) for the pattern in question. I am +1 on making that change before merging.